core/src/main/java/co/aikar/commands/RegisteredCommand.java

```diff
@@ -226,8 +226,13 @@ void handleException(CommandIssuer sender, List<String> args, Exception e) {
             if (requiresInput && remainingRequired > 0) {
                 remainingRequired--;
             }
+            String parameterPermission = parameter.getPermission();
             if (args.isEmpty() && !(isLast && type == String[].class)) {
                 if (allowOptional && parameter.getDefaultValue() != null) {
+                    if (parameterPermission != null && !parameterPermission.isEmpty() && !sender.hasPermission(parameterPermission)) {
```

Comment:
I don't think we need to check if it has a default? we're ultimately saying we're only giving you the permission to supply your own alternative value if you have the permission, else you're forced to use null/default.

if a param was annotated with a default, this would mandate the permission check on every command.